test/e2e/specs/infrastructure/notification__actions.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * @group calypso-pr
+ */
+
+import {
+	DataHelper,
+	NavbarComponent,
+	NotificationsComponent,
+	RestAPIClient,
+	ReaderResponse,
+	SecretsManager,
+	TestAccount,
+	NewCommentResponse,
+} from '@automattic/calypso-e2e';
+import { Page, Browser } from 'playwright';
+
+declare const browser: Browser;
+
+describe( DataHelper.createSuiteTitle( 'Notifications' ), function () {
+	const commentingUser = 'commentingUser';
+	const notificationsUser = 'notificationsUser';
+	const comment = DataHelper.getRandomPhrase() + ' notification-actions-spec';
+
+	let notificationsComponent: NotificationsComponent;
+	let restAPIClient: RestAPIClient;
+	let siteID: number;
+	let commentID: number;
+
+	describe( `Leave a comment as ${ commentingUser }`, function () {
+		let postID: number;
+
+		beforeAll( async function () {
+			restAPIClient = new RestAPIClient( SecretsManager.secrets.testAccounts.commentingUser );
+		} );
+
+		it( 'Get the latest post in reader stream', async function () {
+			const response: ReaderResponse = await restAPIClient.getReaderFeed();
+			siteID = response.posts[ 0 ].site_ID;
+			postID = response.posts[ 0 ].ID;
+		} );
+
+		it( 'Leave a comment on the post', async function () {
+			const response: NewCommentResponse = await restAPIClient.createComment(
+				siteID,
+				postID,
+				comment
+			);
+			commentID = response.ID;
+		} );
+	} );
```

Comment:
**style:non-blocking**: The way you have laid out the setup process is perfectly fine. 

For other specs in the repo, the tendency is to have a `beforeAll` at the top level that would perform the required setup, only assigning `describe` blocks and `it` steps to actual test steps.

eg.

```
describe( Some test suite )
    beforeAll()
         log in
         leave a comment on the post
    
    describe ( User is able to see comment )
         it ( test step 1...)
```

Again, there's nothing wrong with giving the test setup its own `describe` and `it` steps, though at the expense of adding to the step count reported in TeamCity and Allure.

Reply:
Makes sense! I'd rather be more in line with the other specs, so I updated the test to have a top level `beforeAll` like you suggested. Thanks!